ios/RCTMapboxGL/RCTMapboxAnnotation.m

```diff
@@ -17,4 +17,14 @@
 @implementation RCTMapboxAnnotation {
 }
 
+-(NSString *)reuseIdentifier {
+    return self.id;
+}
+
+-(void)layoutSubviews {
+    [super layoutSubviews];
+    [self.map restoreAnnotationPosition:self.id];
```

Comment:
@1ec5 Unfortunately it is not possible. When layoutSubviews is called, position is already changed by internal react native stuff (when JS changes size of that react view, for example).